pkg/deploy/registry/deployconfig/etcd/etcd.go

```diff
@@ -46,3 +75,150 @@ func NewStorage(s storage.Interface) *REST {
+	if err != nil {
+		return nil, errors.NewNotFound("scale", name)
+	}
+
+	// TODO(directxman12): this is going to be a bit out of sync, since we are calculating it
+	// here and not as part of the deploymentconfig loop -- is there a better way of doing it?
+	totalReplicas, err := r.replicasForDeploymentConfig(deploymentConfig.Namespace, deploymentConfig.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	var specReplicas int
+
+	// current replicas reflects either the scale of the current deployment,
+	// or the scale of the RC template if no current deployment exists
+	controller, err := r.rcNamespacer.ReplicationControllers(deploymentConfig.Namespace).Get(util.LatestDeploymentNameForConfig(deploymentConfig))
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return nil, err
+		}
+
+		specReplicas = deploymentConfig.Template.ControllerTemplate.Replicas
+	} else {
+		specReplicas = controller.Spec.Replicas
+	}
+
+	return &extensions.Scale{
+		ObjectMeta: kapi.ObjectMeta{
+			Name:              name,
+			Namespace:         deploymentConfig.Namespace,
+			CreationTimestamp: deploymentConfig.CreationTimestamp,
+		},
+		Spec: extensions.ScaleSpec{
+			Replicas: specReplicas,
+		},
+		Status: extensions.ScaleStatus{
+			Replicas: totalReplicas,
+			Selector: deploymentConfig.Template.ControllerTemplate.Selector,
+		},
+	}, nil
+}
+
+// Update scales the DeploymentConfig for the given Scale subresource, returning the updated Scale.
+func (r *ScaleREST) Update(ctx kapi.Context, obj runtime.Object) (runtime.Object, bool, error) {
+	if obj == nil {
+		return nil, false, errors.NewBadRequest(fmt.Sprintf("nil update passed to Scale"))
+	}
+	scale, ok := obj.(*extensions.Scale)
+	if !ok {
+		return nil, false, errors.NewBadRequest(fmt.Sprintf("wrong object passed to Scale update: %v", obj))
+	}
+	deploymentConfig, err := (*r.registry).GetDeploymentConfig(ctx, scale.Name)
+	if err != nil {
+		return nil, false, errors.NewNotFound("scale", scale.Name)
+	}
+
+	useDC := false
+
+	// this tries to update the current deployment RC first, since updating
+	// Replicas on a DeploymentConfig doesn't do anything after the first deployment
+	// has been made
+	controller, err := r.rcNamespacer.ReplicationControllers(deploymentConfig.Namespace).Get(util.LatestDeploymentNameForConfig(deploymentConfig))
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return nil, false, err
+		}
+
+		useDC = true
+	}
+
+	if deploymentStatus := util.DeploymentStatusFor(controller); deploymentStatus != api.DeploymentStatusComplete {
+		return nil, false, errors.NewConflict("Scale", scale.Name, fmt.Errorf("deployment currently in progress or failed"))
+	}
+
+	// TODO(directxman12): this is going to be a bit out of sync, since we are calculating it
+	// here and not as part of the deploymentconfig loop -- is there a better way of doing it?
+	totalReplicas, err := r.replicasForDeploymentConfig(deploymentConfig.Namespace, deploymentConfig.Name)
```

Comment:
Again, this doesn't make sense when `useDC` is true, right?  

It seems like the case of `useDC` should have early branch and return.